src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -823,7 +823,23 @@ private function syncNode($uuid, $path, $type, $isNewNode, $props = array(), $pr
             $this->syncBinaryData($nodeId, $propsData['binaryData']);
         }
 
-        if (!empty($propsData['references'])) {
+        $needSyncReferences = !empty($propsData['references']);
+        if (!$needSyncReferences) {
+            foreach ($props as $property) {
+                if (true || // maybe we just have to always sync?
+                    $property->isMultiple()
+                    && $property->isModified()
+                    && in_array($property->getType(), array(PropertyType::REFERENCE, PropertyType::WEAKREFERENCE))
+                ) {
+                    // there are no references from this node anymore, which means there is a multivalue
+                    // reference property with an empty array. need to sync to remove previous references
+                    $needSyncReferences = true;
+                    break;
+                }
+            }
+        }
+
+        if ($needSyncReferences) {
```

Comment:
hm, and according to travis we have some issue with converting an array to string, detected on php 5.4+

@lsmith77 if you have any input, please tell.